osu.Game/Online/Chat/MessageNotifier.cs

```diff
@@ -126,10 +127,14 @@ private void checkForMentions(Channel channel, Message message)
         }
 
         /// <summary>
-        /// Checks if <paramref name="message"/> contains <paramref name="username"/>.
+        /// Checks if <paramref name="message"/> mentions <paramref name="username"/>.
         /// This will match against the case where underscores are used instead of spaces (which is how osu-stable handles usernames with spaces).
         /// </summary>
-        private static bool checkContainsUsername(string message, string username) => message.Contains(username, StringComparison.OrdinalIgnoreCase) || message.Contains(username.Replace(' ', '_'), StringComparison.OrdinalIgnoreCase);
+        public static bool checkContainsUsername(string message, string username) {
+            string fullName = Regex.Escape(username);
+            string underscoreName = Regex.Escape(username.Replace(' ', '_'));
+            return new Regex($"\\b({fullName}|{underscoreName})\\b", RegexOptions.IgnoreCase).Matches(message).Count > 0;
```

Comment:
better to use verbatim symbol instead.
```suggestion
            return new Regex($@"\b({fullName}|{underscoreName})\b", RegexOptions.IgnoreCase).Matches(message).Count > 0;
```

Reply:
This seems to be partially applied, you could use `Regex.IsMatch` statically rather than constructing one.